src/index.ts

```diff
@@ -7104,14 +7104,23 @@ async function fetchNetworkAccountFromArchiver(): Promise<WrappedAccount> {
   }[] = []
   for (const archiver of archiverList) {
     try {
-      const res = await axios.get<{ networkAccountHash: string }>(
+      const res = await axios.get<{ networkAccountHash: string,
+        sign: {
+          owner: string,
+          sig: string
+        } 
+      }>(
         `http://${archiver.ip}:${archiver.port}/get-network-account?hash=true`
       )
       if (!res.data) {
         /* prettier-ignore */ nestedCountersInstance.countEvent('network-config-operation', 'failure: did not get network account from archiver private key. Use default configs.')
         throw new Error(`fetchNetworkAccountFromArchiver() from pk:${archiver.publicKey} returned null`)
       }
-
+      const isResponseVerified = shardus.crypto.verify(res.data)
+      const isFromArchiver = archiver.publicKey === res.data.sign.owner
```

Comment:
Swap these two lines, isResponseVerified and isFromArchiver, so that the expensive crypto check is done last. Also throw the error if isFromArchiver is False so the crypto.verify() is not run unnecessarily